lib/train/transports/local.rb

```diff
@@ -50,8 +41,39 @@ def uri
 
       private
 
+      def select_runner(options)
+        if os.windows?
+          # Attempt to use a named pipe but fallback to ShellOut if that fails
+          begin
+            WindowsPipeRunner.new
+          rescue PipeError
+            WindowsShellRunner.new
+          end
+        else
+          GenericRunner.new(self, options)
+        end
+      end
+
+      def force_runner(runner)
+        case runner
+        when 'generic'
+          GenericRunner.new(self, options)
+        when 'windows_pipe'
+          WindowsPipeRunner.new
+        when 'windows_shell'
+          WindowsShellRunner.new
+        end
```

Comment:
We should probably raise an exception if a runner was supplied that we don't support.

Reply:
Good catch, will add an exception.